### PR TITLE
Create Dragen-3.9 pipelines

### PR DESCRIPTION
**Pull request by skanwal**
**Changes include**

- CWL Tool definitions for 
     - dragen-somatic
     - dragen-alignment
     - dragen-transcriptome
     
- CWL workflow definitions for
     - dragen-somatic-pipeline
     - dragen-alignment-pipeline
     - dragen-transcriptome-pipeline
     
- Project and workflow config updates

- Successful workflows have also been registered for catalog

**Note:**

Dragen-transcriptome pipeline hasn't been successfully tested yet due to https://github.com/umccr-illumina/dragen/issues/32

### CODE REVIEW

File: tools/dragen-somatic/3.9.3/dragen-somatic__3.9.3.cwl [hunk 1]
```diff
@@ -0,0 +1,962 @@
...
+s:author:
+    class: s:Person
+    s:name: Sehrish Kanwal
+    s:email: sehrish.kanwal@umccr.org
+
+# ID/Docs
+id: dragen-somatic--3.9.3
+label: dragen-somatic v(3.9.3)
+doc: |
+    Documentation for dragen-somatic v3.9.3
```
Comment (alexiswl) [inline tools/dragen-somatic/3.9.3/dragen-somatic__3.9.3.cwl:21]: Might be worth putting some info in here, for 3.7.5 we have

>  Run tumor-normal dragen somatic pipeline v 3.7.5.
> Workflow takes in two separate lists of object stor version of the fastq_list.csv equivalent
> See the fastq_list_row schema definitions for more information.
> More information on the documentation can be found [here](https://sapac.support.illumina.com/content/dam/illumina-support/help/Illumina_DRAGEN_Bio_IT_Platform_v3_7_1000000141465/Content/SW/Informatics/Dragen/GPipelineSomCom_appDRAG.htm)

**Comment by alexiswl** [comment]
Some differences between the 3.7.5 and 3.9.3 tools (for reference): 

## Dragen alignment Tools
```
4d3
< 
12d10
< 
16,19c14,15
<     s:name: Alexis Lucattini
<     s:email: Alexis.Lucattini@umccr.org
<     s:identifier: https://orcid.org/0000-0001-9754-647X
< 
---
>     s:name: Sehrish Kanwal
>     s:email: sehrish.kanwal@umccr.org
22,23c18,19
< id: dragen-alignment--3.7.5
< label: dragen-alignment v(3.7.5)
---
> id: dragen-alignment--3.9.3
> label: dragen-alignment v(3.9.3)
25,26c21
<     Documentation for dragen-alignment v3.7.5
< 
---
>     Documentation for dragen-alignment v3.9.3
27a23
> # ILMN Resources Guide: https://support-docs.illumina.com/SW/ICA/Content/SW/ICA/RequestResources.htm
38c34
<         dockerPull: "699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:3.7.5"
---
>         dockerPull: "699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:3.9.3"
```

## Dragen somatic tools
```
18,19c18,19
< id: dragen-somatic--3.7.5
< label: dragen-somatic v(3.7.5)
---
> id: dragen-somatic--3.9.3
> label: dragen-somatic v(3.9.3)
21,25c21
<   Run tumor-normal dragen somatic pipeline
<   v 3.7.5.
<   Workflow takes in two separate lists of object stor version of the fastq_list.csv equivalent
<   See the fastq_list_row schema definitions for more information.
<   More information on the documentation can be found [here](https://sapac.support.illumina.com/content/dam/illumina-support/help/Illumina_DRAGEN_Bio_IT_Platform_v3_7_1000000141465/Content/SW/Informatics/Dragen/GPipelineSomCom_appDRAG.htm)
---
>     Documentation for dragen-somatic v3.9.3
26a23
> # ILMN Resources Guide: https://support-docs.illumina.com/SW/ICA/ICA_CLI/Content/SW/ICA/IAPWES_RequestResources.htm
32a30,31
>         coresMin: 2
>         ramMin: 4000
34c33
<         dockerPull: "699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:3.7.5"
---
>         dockerPull: "699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:3.9.3"
303a303,401
>   # Structural Variant Caller Options
>   sv_call_regions_bed:
>     label: sv call regions bed
>     doc: |
>       Specifies a BED file containing the set of regions to call. 
>     type: File?
>     inputBinding:
>       prefix: "--sv-call-regions-bed"
>   sv_region:
>     label: sv region
>     doc: |
>       Limit the analysis to a specified region of the genome for debugging purposes. 
>       This option can be specified multiple times to build a list of regions. 
>       The value must be in the format “chr:startPos-endPos”.. 
>     type: string?
>     inputBinding:
>       prefix: "--sv-region"
>       valueFrom: "$(self.toString())"
>   sv_exome:
>     label: sv exome
>     doc: |
>       Set to true to configure the variant caller for targeted sequencing inputs, 
>       which includes disabling high depth filters. 
>       In integrated mode, the default is to autodetect targeted sequencing input, 
>       and in standalone mode the default is false.
>     type: boolean?
>     inputBinding:
>       prefix: "--sv-exome"
>       valueFrom: "$(self.toString())"
>   sv_output_contigs:
>     label: sv output contigs
>     doc: |
>       Set to true to have assembled contig sequences output in a VCF file. The default is false.
>     type: boolean?
>     inputBinding:
>       prefix: "--sv-output-contigs"
>       valueFrom: "$(self.toString())"
>   sv_forcegt_vcf:
>     label: sv forcegt vcf
>     doc: |
>       Specify a VCF of structural variants for forced genotyping. The variants are scored and emitted 
>       in the output VCF even if not found in the sample data. 
>       The variants are merged with any additional variants discovered directly from the sample data.
>     type: File?
>     inputBinding:
>       prefix: "--sv-forcegt-vcf"
>   sv_discovery:
>     label: sv discovery
>     doc: |
>       Enable SV discovery. This flag can be set to false only when --sv-forcegt-vcf is used. 
>       When set to false, SV discovery is disabled and only the forced genotyping input variants 
>       are processed. The default is true.
>     type: boolean?
>     inputBinding:
>       prefix: "--sv-discovery"
>       valueFrom: "$(self.toString())"
>   sv_se_overlap_pair_evidence:
>     label: sv use overlap pair evidence
>     doc: |
>       Allow overlapping read pairs to be considered as evidence. 
>       By default, DRAGEN uses autodetect on the fraction of overlapping read pairs if <20%.
>     type: boolean?
>     inputBinding:
>       prefix: "--sv-use-overlap-pair-evidence"
>       valueFrom: "$(self.toString())"
>   sv_somatic_ins_tandup_hotspot_regions_bed:
>     label: sv somatic ins tandup hotspot regions bed
>     doc: |
>       Specify a BED of ITD hotspot regions to increase sensitivity for calling ITDs in somatic variant analysis.
>       By default, DRAGEN SV automatically selects areference-specific hotspots BED file from 
>       /opt/edico/config/sv_somatic_ins_tandup_hotspot_*.bed.
>     type: File?
>     inputBinding:
>       prefix: "--sv-somatic-ins-tandup-hotspot-regions-bed"
>   sv_enable_somatic_ins_tandup_hotspot_regions:
>     label: sv enable somatic ins tandup hotspot regions
>     doc: |
>       Enable or disable the ITD hotspot region input. The default is true in somatic variant analysis.
>     type: boolean?
>     inputBinding:
>       prefix: "--sv-enable-somatic-ins-tandup-hotspot-regions"
>       valueFrom: "$(self.toString())"
>   sv_enable_liquid_tumor_mode:
>     label: sv enable liquid tumor mode
>     doc: |
>       Enable liquid tumor mode. 
>     type: boolean?
>     inputBinding:
>       prefix: "--sv-enable-liquid-tumor-mode"
>       valueFrom: "$(self.toString())"  
>   sv_tin_contam_tolerance:
>     label: sv tin contam tolerance
>     doc: |
>       Set the Tumor-in-Normal (TiN) contamination tolerance level. 
>       You can enter any value between 0–1. The default maximum TiN contamination tolerance is 0.15. 
>     type: float?
>     inputBinding:
>       prefix: "--sv-tin-contam-tolerance"
>   
500c598
<      --vc-tin-contam-tolerance enables liquid tumor mode and allows you to
---
>       vc-tin-contam-tolerance enables liquid tumor mode and allows you to
504a603,610
>   vc_enable_orientation_bias_filter:
>     label: vc enable orientation bias filter
>     type: float?
>     doc: |
>       Enables the orientation bias filter. The default value is false, which means the option is disabled.
>     inputBinding:
>       prefix: --vc-enable-orientation-bias-filter
>       valueFrom: "$(self.toString())"
584c690,693
<   # cnv pipeline
---
>   # cnv pipeline - with this we must also specify one of --cnv-normal-b-allele-vcf, 
>   # --cnv-population-b-allele-vcf, or cnv-use-somatic-vc-baf.
>   # If known, specify the sex of the sample. 
>   # If the sample sex is not specified, the caller attempts to estimate the sample sex from tumor alignments. 
592a702,886
>   cnv_normal_b_allele_vcf:
>     label: cnv normal b allele vcf
>     doc: |
>       Specify a matched normal SNV VCF.
>     type: File?
>     inputBinding:
>       prefix: --cnv-normal-b-allele-vcf
>   cnv_population_b_allele_vcf:
>     label: cnv population b allele vcf
>     doc: |
>       Specify a population SNP catalog.
>     type: File?
>     inputBinding:
>       prefix: --cnv-population-b-allele-vcf
>   cnv_use_somatic_vc_baf:
>     label: cnv use somatic vc baf
>     doc: |
>       If running in tumor-normal mode with the SNV caller enabled, use this option 
>       to specify the germline heterozygous sites. 
>     type: boolean?
>     inputBinding:
>       prefix: --cnv-use-somatic-vc-baf
>       valueFrom: "$(self.toString())"
>   # For more info on following options - see 
>   # https://support-docs.illumina.com/SW/DRAGEN_v39/Content/SW/DRAGEN/SomaticWGSModes.htm#Germline
>   cnv_normal_cnv_vcf:
>     label: cnv normal cnv vcf
>     doc: |
>       Specify germline CNVs from the matched normal sample.
>     type: boolean?
>     inputBinding:
>       prefix: --cnv-normal-cnv-vcf
>       valueFrom: "$(self.toString())"
>   cnv_use_somatic_vc_vaf:
>     label: cnv use somatic vc vaf
>     doc: |
>       Use the variant allele frequencies (VAFs) from the somatic SNVs to help select 
>       the tumor model for the sample. 
>     type: boolean?
>     inputBinding:
>       prefix: --cnv-use-somatic-vc-vaf
>       valueFrom: "$(self.toString())"
>   cnv_somatic_enable_het_calling:
>     label: cnv somatic enable het calling
>     doc: |
>       Enable HET-calling mode for heterogeneous segments.
>     type: boolean?
>     inputBinding:
>       prefix: --cnv-somatic-enable-het-calling
>       valueFrom: "$(self.toString())"
>   # HRD
>   enable_hrd:
>     label: enable hrd
>     doc: |
>       Set to true to enable HRD scoring to quantify genomic instability.
>       Requires somatic CNV calls.
>     type: boolean?
>     inputBinding:
>       prefix: --enable-hrd
>       valueFrom: "$(self.toString())"
>   # QC options
>   qc_coverage_region_1:
>     label: qc coverage region 1
>     doc: |
>       Generates coverage region report using bed file 1.
>     type: File?
>     inputBinding:
>       prefix: --qc-coverage-region-1
>   qc_coverage_region_2:
>     label: qc coverage region 2
>     doc: |
>       Generates coverage region report using bed file 2.
>     type: File?
>     inputBinding:
>       prefix: --qc-coverage-region-2
>   qc_coverage_region_3:
>     label: qc coverage region 3
>     doc: |
>       Generates coverage region report using bed file 3.
>     type: File?
>     inputBinding:
>       prefix: --qc-coverage-region-3
>   qc_coverage_ignore_overlaps:
>     label: qc coverage ignore overlaps
>     doc: |
>       Set to true to resolve all of the alignments for each fragment and avoid double-counting any 
>       overlapping bases. This might result in marginally longer run times. 
>       This option also requires setting --enable-map-align=true.
>     type: boolean?
>     inputBinding:
>       prefix: --qc-coverage-ignore-overlaps
>       valueFrom: "$(self.toString())"
>   # TMB options
>   enable_tmb:
>     label: enable tmb
>     doc: |
>       Enables TMB. If set, the small variant caller, Illumina Annotation Engine, 
>       and the related callability report are enabled.
>     type: boolean?
>     inputBinding:
>       prefix: --enable-tmb
>       valueFrom: "$(self.toString())"
>   tmb_vaf_threshold:
>     label: tmb vaf threshold
>     doc: |
>       Specify the minimum VAF threshold for a variant. Variants that do not meet the threshold are filtered out. 
>       The default value is 0.05.
>     type: float?
>     inputBinding:
>       prefix: --tmb-db-threshold
>   tmb_db_threshold:
>     label: tmb db threshold
>     doc: |
>       Specify the minimum allele count (total number of observations) for an allele in gnomAD or 1000 Genome 
>       to be considered a germline variant.  Variant calls that have the same positions and allele are ignored
>       from the TMB calculation. The default value is 10.
>     type: int?
>     inputBinding:
>       prefix: --tmb-db-threshold
>   # HLA calling
>   enable_hla:
>     label: enable hla
>     doc: |
>       Enable HLA typing by setting --enable-hla flag to true
>     type: boolean?
>     inputBinding:
>       prefix: --enable-hla
>       valueFrom: "$(self.toString())"
>   hla_bed_file:
>     label: hla bed file
>     doc: |
>       Use the HLA region BED input file to specify the region to extract HLA reads from. 
>       DRAGEN HLA Caller parses the input file for regions within the BED file, and then 
>       extracts reads accordingly to align with the HLA allele reference.
>     type: File?
>     inputBinding:
>       prefix: --hla-bed-file
>   hla_reference_file:
>     label: hla reference file
>     doc: |
>       Use the HLA allele reference file to specify the reference alleles to align against.
>       The input HLA reference file must be in FASTA format and contain the protein sequence separated into exons.
>       If --hla-reference-file is not specified, DRAGEN uses hla_classI_ref_freq.fasta from /opt/edico/config/. 
>       The reference HLA sequences are obtained from the IMGT/HLA database.
>     type: File?
>     inputBinding:
>       prefix: --hla-reference-file
>   hla_allele_frequency_file:
>     label: hla allele frequency file
>     doc: |
>       Use the population-level HLA allele frequency file to break ties if one or more HLA allele produces the same or similar results.
>       The input HLA allele frequency file must be in CSV format and contain the HLA alleles and the occurrence frequency in population.
>       If --hla-allele-frequency-file is not specified, DRAGEN automatically uses hla_classI_allele_frequency.csv from /opt/edico/config/. 
>       Population-level allele frequencies can be obtained from the Allele Frequency Net database.
>     type: File?
>     inputBinding:
>       prefix: --hla-allele-frequency-file
>   hla_tiebreaker_threshold:
>     label: hla tiebreaker threshold
>     doc: |
>       If more than one allele has a similar number of reads aligned and there is not a clear indicator for the best allele, 
>       the alleles are considered as ties. The HLA Caller places the tied alleles into a candidate set for tie breaking based 
>       on the population allele frequency. If an allele has more than the specified fraction of reads aligned (normalized to 
>       the top hit), then the allele is included into the candidate set for tie breaking. The default value is 0.97.
>     type: float?
>     inputBinding:
>       prefix: --hla-tiebreaker-threshold
>   hla_zygosity_threshold:
>     label: hla zygosity threshold
>     doc: |
>       If the minor allele at a given locus has fewer reads mapped than a fraction of the read count of the major allele, 
>       then the HLA Caller infers homozygosity for the given HLA-I gene. You can use this option to specify the fraction value. 
>       The default value is 0.15.
>     type: float?
>     inputBinding:
>       prefix: --hla zygosity threshold
>   hla_min_reads:
>     label: hla min reads
>     doc: |
>       Set the minimum number of reads to align to HLA alleles to ensure sufficient coverage and perform HLA typing. 
>       The default value is 1000 and suggested for WES samples. If using samples with less coverage, you can use a 
>       lower threshold value.
>     type: int?
>     inputBinding:
>       prefix: --hla-min-reads
```

## Dragen transcriptome tools (3.8.4 v 3.9.3)
```
18,19c18,19
< id: dragen-transcriptome--3.8.4
< label: dragen-transcriptome v(3.8.4)
---
> id: dragen-transcriptome--3.9.3
> label: dragen-transcriptome v(3.9.3)
21c21
<     Documentation for dragen-transcriptome v3.8.4
---
>     Documentation for dragen-transcriptome v3.9.3
31c31
<         dockerPull: "699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:3.8.4"
---
>         dockerPull: "699120554104.dkr.ecr.us-east-1.amazonaws.com/public/dragen:3.9.3"
```

**Comment by alexiswl** [comment]
Some differences between the 3.7.5 and 3.9.3 workflows (for reference):

## Dragen wgs qc pipeline
```
17,18c17,18
< id: dragen-wgs-qc-pipeline--3.7.5
< label: dragen-wgs-qc-pipeline v(3.7.5)
---
> id: dragen-wgs-qc-pipeline--3.9.3
> label: dragen-wgs-qc-pipeline v(3.9.3)
20c20
<     Documentation for dragen-alignment-pipeline v3.7.5
---
>     Documentation for dragen-wgs-qc-pipeline v3.9.3
107c107
<     run: ../../../workflows/dragen-alignment-pipeline/3.7.5/dragen-alignment-pipeline__3.7.5.cwl
---
>     run: ../../../workflows/dragen-alignment-pipeline/3.9.3/dragen-alignment-pipeline__3.9.3.cwl
164d163
< 
```

## Dragen somatic pipeline
```
17,18c17,18
< id: dragen-somatic-pipeline--3.7.5
< label: dragen-somatic-pipeline v(3.7.5)
---
> id: dragen-somatic-pipeline--3.9.3
> label: dragen-somatic-pipeline v(3.9.3)
21c21
<   v 3.7.5.
---
>   v 3.9.3.
24c24
<   More information on the documentation can be found [here](https://sapac.support.illumina.com/content/dam/illumina-support/help/Illumina_DRAGEN_Bio_IT_Platform_v3_7_1000000141465/Content/SW/Informatics/Dragen/GPipelineSomCom_appDRAG.htm)
---
>   More information on the documentation can be found [here](https://support-docs.illumina.com/SW/DRAGEN_v39/Content/SW/FrontPages/DRAGEN.htm)
123a124,192
>   # Structural Variant Caller Options
>   sv_call_regions_bed:
>     label: sv call regions bed
>     doc: |
>       Specifies a BED file containing the set of regions to call. 
>     type: File?
>   sv_region:
>     label: sv region
>     doc: |
>       Limit the analysis to a specified region of the genome for debugging purposes. 
>       This option can be specified multiple times to build a list of regions. 
>       The value must be in the format “chr:startPos-endPos”.. 
>     type: string?
>   sv_exome:
>     label: sv exome
>     doc: |
>       Set to true to configure the variant caller for targeted sequencing inputs, 
>       which includes disabling high depth filters. 
>       In integrated mode, the default is to autodetect targeted sequencing input, 
>       and in standalone mode the default is false.
>     type: boolean?
>   sv_output_contigs:
>     label: sv output contigs
>     doc: |
>       Set to true to have assembled contig sequences output in a VCF file. The default is false.
>     type: boolean?
>   sv_forcegt_vcf:
>     label: sv forcegt vcf
>     doc: |
>       Specify a VCF of structural variants for forced genotyping. The variants are scored and emitted 
>       in the output VCF even if not found in the sample data. 
>       The variants are merged with any additional variants discovered directly from the sample data.
>     type: File?
>   sv_discovery:
>     label: sv discovery
>     doc: |
>       Enable SV discovery. This flag can be set to false only when --sv-forcegt-vcf is used. 
>       When set to false, SV discovery is disabled and only the forced genotyping input variants 
>       are processed. The default is true.
>     type: boolean?
>   sv_se_overlap_pair_evidence:
>     label: sv use overlap pair evidence
>     doc: |
>       Allow overlapping read pairs to be considered as evidence. 
>       By default, DRAGEN uses autodetect on the fraction of overlapping read pairs if <20%.
>     type: boolean?
>   sv_somatic_ins_tandup_hotspot_regions_bed:
>     label: sv somatic ins tandup hotspot regions bed
>     doc: |
>       Specify a BED of ITD hotspot regions to increase sensitivity for calling ITDs in somatic variant analysis.
>       By default, DRAGEN SV automatically selects areference-specific hotspots BED file from 
>       /opt/edico/config/sv_somatic_ins_tandup_hotspot_*.bed.
>     type: File?
>   sv_enable_somatic_ins_tandup_hotspot_regions:
>     label: sv enable somatic ins tandup hotspot regions
>     doc: |
>       Enable or disable the ITD hotspot region input. The default is true in somatic variant analysis.
>     type: boolean?
>   sv_enable_liquid_tumor_mode:
>     label: sv enable liquid tumor mode
>     doc: |
>       Enable liquid tumor mode. 
>     type: boolean? 
>   sv_tin_contam_tolerance:
>     label: sv tin contam tolerance
>     doc: |
>       Set the Tumor-in-Normal (TiN) contamination tolerance level. 
>       You can enter any value between 0–1. The default maximum TiN contamination tolerance is 0.15. 
>     type: float?
341a411,545
>   cnv_normal_b_allele_vcf:
>     label: cnv normal b allele vcf
>     doc: |
>       Specify a matched normal SNV VCF.
>     type: File?
>   cnv_population_b_allele_vcf:
>     label: cnv population b allele vcf
>     doc: |
>       Specify a population SNP catalog.
>     type: File?
>   cnv_use_somatic_vc_baf:
>     label: cnv use somatic vc baf
>     doc: |
>       If running in tumor-normal mode with the SNV caller enabled, use this option 
>       to specify the germline heterozygous sites. 
>     type: boolean?
>   # For more info on following options - see 
>   # https://support-docs.illumina.com/SW/DRAGEN_v39/Content/SW/DRAGEN/SomaticWGSModes.htm#Germline
>   cnv_normal_cnv_vcf:
>     label: cnv normal cnv vcf
>     doc: |
>       Specify germline CNVs from the matched normal sample.
>     type: boolean?
>   cnv_use_somatic_vc_vaf:
>     label: cnv use somatic vc vaf
>     doc: |
>       Use the variant allele frequencies (VAFs) from the somatic SNVs to help select 
>       the tumor model for the sample. 
>     type: boolean?
>   cnv_somatic_enable_het_calling:
>     label: cnv somatic enable het calling
>     doc: |
>       Enable HET-calling mode for heterogeneous segments. 
>     type: boolean?
>   # HRD
>   enable_hrd:
>     label: enable hrd
>     doc: |
>       Set to true to enable HRD scoring to quantify genomic instability.
>       Requires somatic CNV calls.
>     type: boolean?
>   # QC options
>   qc_coverage_region_1:
>     label: qc coverage region 1
>     doc: |
>       Generates coverage region report using bed file 1.
>     type: File?
>   qc_coverage_region_2:
>     label: qc coverage region 2
>     doc: |
>       Generates coverage region report using bed file 2.
>     type: File?
>   qc_coverage_region_3:
>     label: qc coverage region 3
>     doc: |
>       Generates coverage region report using bed file 3.
>     type: File?
>   qc_coverage_ignore_overlaps:
>     label: qc coverage ignore overlaps
>     doc: |
>       Set to true to resolve all of the alignments for each fragment and avoid double-counting any 
>       overlapping bases. This might result in marginally longer run times. 
>       This option also requires setting --enable-map-align=true.
>     type: boolean?
>   # TMB options
>   enable_tmb:
>     label: enable tmb
>     doc: |
>       Enables TMB. If set, the small variant caller, Illumina Annotation Engine, 
>       and the related callability report are enabled.
>     type: boolean?
>   tmb_vaf_threshold:
>     label: tmb vaf threshold
>     doc: |
>       Specify the minimum VAF threshold for a variant. Variants that do not meet the threshold are filtered out. 
>       The default value is 0.05.
>     type: float?
>   tmb_db_threshold:
>     label: tmb db threshold
>     doc: |
>       Specify the minimum allele count (total number of observations) for an allele in gnomAD or 1000 Genome 
>       to be considered a germline variant.  Variant calls that have the same positions and allele are ignored
>       from the TMB calculation. The default value is 10.
>     type: int?
>   # HLA calling
>   enable_hla:
>     label: enable hla
>     doc: |
>       Enable HLA typing by setting --enable-hla flag to true
>     type: boolean?
>   hla_bed_file:
>     label: hla bed file
>     doc: |
>       Use the HLA region BED input file to specify the region to extract HLA reads from. 
>       DRAGEN HLA Caller parses the input file for regions within the BED file, and then 
>       extracts reads accordingly to align with the HLA allele reference.
>     type: File?
>   hla_reference_file:
>     label: hla reference file
>     doc: |
>       Use the HLA allele reference file to specify the reference alleles to align against.
>       The input HLA reference file must be in FASTA format and contain the protein sequence separated into exons.
>       If --hla-reference-file is not specified, DRAGEN uses hla_classI_ref_freq.fasta from /opt/edico/config/. 
>       The reference HLA sequences are obtained from the IMGT/HLA database.
>     type: File?
>   hla_allele_frequency_file:
>     label: hla allele frequency file
>     doc: |
>       Use the population-level HLA allele frequency file to break ties if one or more HLA allele produces the same or similar results.
>       The input HLA allele frequency file must be in CSV format and contain the HLA alleles and the occurrence frequency in population.
>       If --hla-allele-frequency-file is not specified, DRAGEN automatically uses hla_classI_allele_frequency.csv from /opt/edico/config/. 
>       Population-level allele frequencies can be obtained from the Allele Frequency Net database.
>     type: File?
>   hla_tiebreaker_threshold:
>     label: hla tiebreaker threshold
>     doc: |
>       If more than one allele has a similar number of reads aligned and there is not a clear indicator for the best allele, 
>       the alleles are considered as ties. The HLA Caller places the tied alleles into a candidate set for tie breaking based 
>       on the population allele frequency. If an allele has more than the specified fraction of reads aligned (normalized to 
>       the top hit), then the allele is included into the candidate set for tie breaking. The default value is 0.97.
>     type: float?
>   hla_zygosity_threshold:
>     label: hla zygosity threshold
>     doc: |
>       If the minor allele at a given locus has fewer reads mapped than a fraction of the read count of the major allele, 
>       then the HLA Caller infers homozygosity for the given HLA-I gene. You can use this option to specify the fraction value. 
>       The default value is 0.15.
>     type: float?
>   hla_min_reads:
>     label: hla min reads
>     doc: |
>       Set the minimum number of reads to align to HLA alleles to ensure sufficient coverage and perform HLA typing. 
>       The default value is 1000 and suggested for WES samples. If using samples with less coverage, you can use a 
>       lower threshold value.
>     type: int?
409a614,636
>       # Structural Variant Caller Options
>       sv_call_regions_bed:
>         source: sv_call_regions_bed
>       sv_region:
>         source: sv_region
>       sv_exome:
>         source: sv_exome
>       sv_output_contigs:
>         source: sv_output_contigs
>       sv_forcegt_vcf:
>         source: sv_forcegt_vcf
>       sv_discovery:
>         source: sv_discovery
>       sv_se_overlap_pair_evidence:
>         source: sv_se_overlap_pair_evidence
>       sv_somatic_ins_tandup_hotspot_regions_bed:
>         source: sv_somatic_ins_tandup_hotspot_regions_bed
>       sv_enable_somatic_ins_tandup_hotspot_regions:
>         source: sv_enable_somatic_ins_tandup_hotspot_regions
>       sv_enable_liquid_tumor_mode:
>         source: sv_enable_liquid_tumor_mode
>       sv_tin_contam_tolerance:
>         source: sv_tin_contam_tolerance
469a697
>       #cnv options
472,473c700,741
<       lic_instance_id_location:
<         source: lic_instance_id_location
---
>       cnv_normal_b_allele_vcf:
>         source: cnv_normal_b_allele_vcf
>       cnv_population_b_allele_vcf:
>         source: cnv_population_b_allele_vcf
>       cnv_use_somatic_vc_baf:
>         source: cnv_use_somatic_vc_baf
>       cnv_normal_cnv_vcf:
>         source: cnv_normal_cnv_vcf
>       cnv_use_somatic_vc_vaf:
>         source: cnv_use_somatic_vc_vaf
>       cnv_somatic_enable_het_calling:
>         source: cnv_somatic_enable_het_calling
>       enable_hrd:
>         source: enable_hrd
>       qc_coverage_region_1:
>         source: qc_coverage_region_1
>       qc_coverage_region_2:
>         source: qc_coverage_region_2
>       qc_coverage_region_3:
>         source: qc_coverage_region_3
>       qc_coverage_ignore_overlaps:
>         source: qc_coverage_ignore_overlaps
>       enable_tmb:
>         source: enable_tmb
>       tmb_vaf_threshold:
>         source: tmb_vaf_threshold
>       tmb_db_threshold:
>         source: tmb_db_threshold
>       enable_hla:
>         source: enable_hla
>       hla_bed_file:
>         source: hla_bed_file
>       hla_reference_file:
>         source: hla_reference_file
>       hla_allele_frequency_file:
>         source: hla_allele_frequency_file
>       hla_tiebreaker_threshold:
>         source: hla_tiebreaker_threshold
>       hla_zygosity_threshold:
>         source: hla_zygosity_threshold
>       hla_min_reads:
>         source: hla_min_reads
481c749
<     run: ../../../tools/dragen-somatic/3.7.5/dragen-somatic__3.7.5.cwl
---
>     run: ../../../tools/dragen-somatic/3.9.3/dragen-somatic__3.9.3.cwl
520d787
< 
```

## Dragen transcriptome pipeline (3.8.4 v 3.9.3)
```
17,18c17,18
< id: dragen-transcriptome-pipeline--3.8.4
< label: dragen-transcriptome-pipeline v(3.8.4)
---
> id: dragen-transcriptome-pipeline--3.9.3
> label: dragen-transcriptome-pipeline v(3.9.3)
22c22
<   More information on the documentation can be found [here](https://sapac.support.illumina.com/content/dam/illumina-support/help/Illumina_DRAGEN_Bio_IT_Platform_v3_7_1000000141465/Content/SW/Informatics/Dragen/GPipelineVarCal_fDG.htm)
---
>   More information on the documentation can be found [here](https://support-docs.illumina.com/SW/DRAGEN_v39/Content/SW/DRAGEN/TPipelineIntro_fDG.htm)
136c136
<   # Step-2: Run dragen transcriptome workflow
---
>   # Step-2: Run Dragen transcriptome workflow
169c169
<     run: ../../../tools/dragen-transcriptome/3.8.4/dragen-transcriptome__3.8.4.cwl
---
>     run: ../../../tools/dragen-transcriptome/3.9.3/dragen-transcriptome__3.9.3.cwl

```

File: tools/dragen-somatic/3.9.3/dragen-somatic__3.9.3.cwl (hunk 1 above)
Comment (skanwal) [reply tools/dragen-somatic/3.9.3/dragen-somatic__3.9.3.cwl]: Fixed in https://github.com/umccr/cwl-ica/pull/22/commits/30f53966ef9228f9dec8b9e55935ce6c06d17c09

**Review by alexiswl — APPROVED** [review]

**Comment by alexiswl** [comment]
Sync hung when trying to install conda, will investigate if mamba might be a more reliable option

**Comment by skanwal** [comment]
Can we merge anyway?

**Comment by alexiswl** [comment]
Yeah sure! Will just mean that the 3.9 pipelines won't be in the catalogue for now.

**Pull request merged**